util/binfmt_misc/detect.go

```diff
@@ -14,19 +15,56 @@ func SupportedPlatforms() []string {
 	once.Do(func() {
 		def := platforms.DefaultString()
 		arr = append(arr, def)
-
-		if p := "linux/amd64"; def != p && amd64Supported() {
+		if p := "linux/amd64"; def != p && (amd64Supported() == nil) {
```

Comment:
parenthesis are not needed in these checks

Reply:
agreed, but parenthesis here make it easier to read, anyway, I will remove it.